docs/book/src/quinn/certificate.md

```diff
@@ -129,14 +129,10 @@ This is the only thing you need to do for your server to be secured.
 **Configure Client**
 
 ```rust
-let mut crypto = rustls::ClientConfig::builder()
-    .with_safe_defaults()
-    .with_single_cert(vec![cert]), key)?
-    .with_no_client_auth();
-let client_config = ClientConfig::new(Arc::new(crypto));
+let client_config = ClientConfig::with_native_roots();
```

Comment:
This only works with hyper-rustls, I think? rustls itself doesn't supply `ClientConfig::with_native_roots()`. Instead, we should probably have some simplified version of what that code uses internally:

```rust
let mut roots = rustls::RootCertStore::empty();
        let mut valid_count = 0;
        let mut invalid_count = 0;

        for cert in rustls_native_certs::load_native_certs().expect("could not load platform certs")
        {
            let cert = rustls::Certificate(cert.0);
            match roots.add(&cert) {
                Ok(_) => valid_count += 1,
                Err(err) => {
                    crate::log::trace!("invalid cert der {:?}", cert.0);
                    crate::log::debug!("certificate parsing failed: {:?}", err);
                    invalid_count += 1
                }
            }
        }
```

Reply:
I believe this is `quinn::ClientConfig`, which does have such a helper, not `rustls::ClientConfig`.

Reply:
Ah, sorry, ok.